app/models/api/area.rb

```diff
@@ -43,6 +43,10 @@ def self.all_by_area_code
     end
   end
 
+  def self.derived_datasets
+    all_by_area_code.sort_by { |area_code, _| area_code }.map { |_, area| area }.find_all { |area| area.is_derived? }
```

Comment:
I agree with your first version :+1:
The second is dangerous since it goes around the cache... but maybe we should just cache "all" instead of "all_by_area_code"... after thinking about it: We should stick with the indexed cache, otherwise the finder method has to do a linear search